src/main/java/edu/tamu/app/service/manager/GitHubMilestoneService.java

```diff
@@ -0,0 +1,115 @@
+import edu.tamu.app.model.ManagementService;
+
+public class GitHubMilestoneService extends AbstractGitHubService {
+
+    public GitHubMilestoneService(final ManagementService managementService) throws IOException {
+        super(managementService);
+    }
+
+    @Override
+    public List<Sprint> getActiveSprintsByScopeId(final String scopeId) throws Exception {
+        logger.info("Fetching active sprints for remote project with scope id " + scopeId);
+        List<Sprint> activeSprints = new ArrayList<Sprint>();
+        GHRepository repo = github.getRepositoryById(scopeId);
+        List<GHProject> projects = repo.listProjects(ProjectStateFilter.OPEN).asList();
+        for (GHProject project : projects) {
+            String sprintId = String.valueOf(project.getId());
+            String projectName = repo.getName();
+            Map<String, List<Card>> partitionedCards = getCards(project);
+            for (Entry<String, List<Card>> partition : partitionedCards.entrySet()) {
+                activeSprints.add(new Sprint(sprintId, partition.getKey(), projectName, partition.getValue()));
+            }
+        }
+        return activeSprints;
+    }
+
+    @Override
+    public List<Sprint> getAdditionalActiveSprints() throws Exception {
+        GHOrganization organization = github.getOrganization(ORGANIZATION);
+        List<GHProject> projects = organization.listProjects(ProjectStateFilter.OPEN).asList();
+        List<Sprint> sprints = new ArrayList<Sprint>();
+        for (GHProject project : projects) {
+            String sprintId = String.valueOf(project.getId());
+            Map<String, List<Card>> partitionedCards = getCards(project);
+            String productName = String.format("%s - %s", organization.getName(), project.getName());
+            int count = 0;
+            for (Entry<String, List<Card>> partition : partitionedCards.entrySet()) {
+                sprints.add(new Sprint(sprintId + "-" + count, partition.getKey(), productName, partition.getValue()));
+                count++;
+            }
+        }
+        return sprints;
+    }
+
+    private Map<String, List<Card>> getCards(GHProject project) throws IOException {
+        Map<String, List<Card>> cardsByMilestone = new HashMap<>();
+        for (GHProjectColumn column : project.listColumns().asList()) {
+            List<GHProjectCard> projectCards = column.listCards().asList();
+            Map<Long, GHIssue> cardContents = new HashMap<>();
+            for (GHProjectCard card : projectCards) {
+                cardContents.put(card.getId(), card.getContent());
+            }
+            Map<GHMilestone, List<GHProjectCard>> partitionedCards = projectCards.stream()
+                // Card without contents is a note
+                .filter(c -> cardContents.get(c.getId()) != null)
+                // Card without a milestone is not on the sprint
+                .filter(c -> cardContents.get(c.getId()).getMilestone() != null)
```

Comment:
Should make sure milestone contains “sprint” in its name. Otherwise, issues on a board under a general milestone will be turned into a sprint.

Reply:
I'm checking for that on line 102, but It would be better to do it here before it's processed.